app/models/manageiq/providers/base_manager/event_catcher/runner.rb

```diff
@@ -84,10 +125,24 @@ def monitor_events
     raise NotImplementedError, _("must be implemented in subclass")
   end
 
-  def process_event(_event)
+  # This method has been refactored to go through two steps, namely filetered? and queue_event.
+  # Therefore every subclass should override filtered? and queue_event
+  #
+  # For historical reason existing providers still directly implement process_event
+  # They should be eventually refactored following the example of VMWare provider.
+  def process_event(event)
+    queue_event(event) unless filtered?(event)
+  end
+
+  def queue_event(_event)
     raise NotImplementedError, _("must be implemented in subclass")
```

Comment:
@durandom I'll leave as is. Commit 787ddafe changed all exception texts. Are you going to revert them all?